match/lib/match/nuke.rb

```diff
@@ -76,6 +80,14 @@ def run(params, type: nil)
       end
     end
 
+    # Be smart about optional values here
+    # Depending on the storage mode, different vlaues are required
+    def update_optional_values_depending_on_storage_type(params)
+      if params[:storage_mode] != "git"
+        params.option_for_key(:git_url).optional = true
```

Comment:
+1 to that. 

I've thought that maybe one way to get around this is to set `available_options` to the common denominator of all storage modes (e.g. `type`, `app_identifier`, and so on) and then have the storage implementation add the options it needs to the list when it is configured. 

I think the thing that's not nice about that is the user doesn't get the options on `--help` in the list, but rather has to consult the docs out of bounds to get that information. On the other hand it might be confusing when to set what parameters depending on the storage you're using, so they'd need to consult the docs anyway. That'd be better still than confusing/overloaded or extra parameter names

Reply:
I like Manu's idea

Reply:
We can even use the `storage_mode` verify block to call the implementation's method that appends the options

Reply:
👍 Sounds good to me. I'd add something like a `base_options` method to match's `options.rb` and then in the `storage_mode` block set the available options to `base_options + GoogleCloudStorage.available_options`